build.sbt

```diff
@@ -47,123 +46,109 @@ lazy val commonSettings = Seq(
   pomExtra := pomInfo,
   resolvers += Opts.resolver.mavenLocalFile,
   autoAPIMappings := true,
-  libraryDependencies ++= excludeServlet(Seq(
-    "org.scalaz" %% "scalaz-core" % "7.1.3",
-    "org.mortbay.jetty" % "servlet-api" % "3.0.20100224" % "provided",
-    "org.apache.hadoop" % "hadoop-common" % hadoopVer % "provided",
-    "org.apache.hadoop" % "hadoop-mapreduce-client-core" % hadoopVer % "provided",
-    "org.apache.parquet" % "parquet-column" % parquetVer,
-    "org.apache.parquet" % "parquet-hadoop" % parquetVer
-  )),
   libraryDependencies ++= Seq(
-    "io.circe" %% "circe-core",
-    "io.circe" %% "circe-generic",
-    "io.circe" %% "circe-parser"
-  ).map(_ % circeVersion),
-  dependencyOverrides += "org.slf4j" % "slf4j-log4j12" % "1.7.12",
-  dependencyOverrides += "org.slf4j" % "slf4j-api" % "1.7.12",
+    "org.apache.hadoop" % "hadoop-common" % hadoopVer % "provided,test",
+    "org.slf4j" % "slf4j-api" % "1.7.24" % "test",
+    "org.slf4j" % "jcl-over-slf4j" % "1.7.24" % "test",
+    "org.slf4j" % "jul-to-slf4j" % "1.7.24" % "test",
+    "org.slf4j" % "log4j-over-slf4j" % "1.7.24" % "test",
+    "ch.qos.logback" % "logback-classic" % "1.2.1" % "test"
+  ),
+  excludeDependencies ++= Seq(
+    // Our logging strategy is to use Logback (logback-classic) which provides
+    // an SLF4J compatible API. We then import as many SLF4J bridges we can
+    // so that every logging library effectively works with our Logback based
+    // logging. Thus, anything outside of this needs to be excluded.
+    SbtExclusionRule(organization = "commons-logging"),
+    SbtExclusionRule(organization = "log4j", name = "log4j"),
+    SbtExclusionRule(organization = "org.slf4j", name = "slf4j-simple"),
+    SbtExclusionRule(organization = "org.slf4j", name = "slf4j-log4j12"),
+    SbtExclusionRule(organization = "org.slf4j", name = "slf4j-jdk14")
+  ),
   resolvers ++= Seq(
     Resolver.typesafeRepo("releases"),
     Resolver.sonatypeRepo("public"),
     Resolver.sonatypeRepo("releases"),
     Resolver.typesafeIvyRepo("releases"),
     "Twitter" at "http://maven.twttr.com/",
     "Bintray" at "https://jcenter.bintray.com/"
-  )
+  ),
+  scalacOptions in ThisBuild ++= Seq("-language:higherKinds")
```

Comment:
I seem to remember you can add an import in the specific file to avoid the warning?